src/hooks/useLocalStorage.tsx

```diff
@@ -0,0 +1,20 @@
+// https://www.daleseo.com/react-hooks-use-web-storage/
+// https://stackoverflow.com/questions/67700374/use-localstorage-getitem-with-typescript
+
+import { ITodoItem } from '../Interface';
+import { useState, useEffect } from 'react';
+
+// 당장은 필요하지 않더라도 재사용이 가능하게끔 짜는 습관 좋은것 같습니다!!
+const useLocalStorage = (key: string, initialState: ITodoItem[]) => {
+    // list state 초기화
+    const [list, setList] = useState<ITodoItem[]>(() => JSON.parse(localStorage.getItem(key) || '') || initialState);
+
+    // list 상태 변화할 때마다 로컬스토리지 동기화
+    useEffect(() => {
+        localStorage.setItem(key, JSON.stringify(list));
+    }, [key, list]);
+
+    return { list, setList };
+};
```

Comment:
```suggestion
const useLocalStorage<T> = (key: string, initialState: T) => {
    // list state 초기화
    const [list, setList] = useState<T>(() => JSON.parse(localStorage.getItem(key) || '') || initialState);

    // list 상태 변화할 때마다 로컬스토리지 동기화
    useEffect(() => {
        localStorage.setItem(key, JSON.stringify(list));
    }, [key, list]);

    return { list, setList };
};
```
이렇게 제네릭으로 hooks를 리팩토링하고, 이 훅이 필요한 컴포넌트에서는,

```ts
const { list, setlist } = useLocalStorage<ITodoItem[]>(null, todoItem)
```
이렇게 쓸수 있겠죠. 이러면 `useLocalStorag`의 범용성이 증가할거 같네요

Reply:
제네릭이 이렇게 사용하는것이었군요..!